### PR TITLE
Add support for capacity providers by using tf 1.3 optional object parmeters

### PR DESCRIPTION
**Pull request by saefty**
- Adds tf support for optional object values with defaults.
- Adds support for capacity providers for FARGATE and FARGATE_SPOT.
- Adds support for deployment circuit breakers.
- This change dropps support for many terraform versions 🚧🔥

### CODE REVIEW

**Review by thisismana — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -52,6 +52,28 @@ variable "appautoscaling_settings" {
   type        = map(any)
 }
 
+variable "capacity_provider_strategy" {
+  default     = null
+  description = "List of capacity provider strategies."
+  type = list(object({
+    capacity_provider = string
+    weight            = optional(string, null)
```
Comment (moritzzimmer) [inline variables.tf:60]: weight is required according to https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_service#weight

File: variables.tf [hunk 2]
```diff
@@ -52,6 +52,28 @@ variable "appautoscaling_settings" {
   type        = map(any)
 }
 
+variable "capacity_provider_strategy" {
+  default     = null
+  description = "List of capacity provider strategies."
```
Comment (moritzzimmer) [inline variables.tf:57]:
```suggestion
  description = "Capacity provider strategies to use for the service. Can be one or more."
```

File: variables.tf [hunk 3]
```diff
@@ -52,6 +52,28 @@ variable "appautoscaling_settings" {
...
+
+variable "deployment_circuit_breaker" {
+  default = {
+    enable   = false
+    rollback = false
+  }
+  description = "Deployment circuit breaker configuration."
+  type = object({
+    enable   = bool
+    rollback = optional(bool, false)
```
Comment (moritzzimmer) [inline variables.tf:73]: this is required according to https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_service#deployment_circuit_breaker, IMO we should keep it that way

**Review by moritzzimmer — CHANGES_REQUESTED** [review]
and please update the docs with the new variables

**Pull request merged**